src/ActiveSpeakerObserver.ts

```diff
@@ -0,0 +1,68 @@
+import { Logger } from './Logger';
+import { EnhancedEventEmitter } from './EnhancedEventEmitter';
+import { RtpObserver } from './RtpObserver';
+import { Producer } from './Producer';
+
+export interface ActiveSpeakerObserverOptions {
+	interval?: number;
+	/**
+	 * Custom application data.
+	 */
+	appData?: any;
+}
+
+export interface ActiveSpeakerObserverActivity {
+	/**
+	 * The producer instance.
+	 */
+	producer: Producer;
```

Comment:
Just a question: will the ActiveSpeakerObserver C++ logic always choose a Producer even if nobody is talking? Imagine that eventually we remove all audio Producers from the ActiveSpeakerObserver. Will it emit some event telling `producer: undefined`? Or will it just NOT emit any event?

Reply:
The first time through the calculation it will pick the first producer/speaker in the map. If the speaker count is 0 then no event is emitted. This is determined in `ActiveSpeakerObserver::CalculateActiveSpeaker()`.